internal/mode/static/nginx/file/manager.go

```diff
@@ -88,10 +89,18 @@ func NewManagerImpl(logger logr.Logger, osFileManager OSFileManager) *ManagerImp
 func (m *ManagerImpl) ReplaceFiles(files []File) error {
 	for _, path := range m.lastWrittenPaths {
 		if err := m.osFileManager.Remove(path); err != nil {
-			return fmt.Errorf("failed to delete file %q: %w", path, err)
+			if os.IsNotExist(err) {
+				m.logger.V(1).Info(
+					"File not found when attempting to replace",
```

Comment:
This will make it consistent with the log line "Deleted file"
```suggestion
					"File not found when attempting to delete",
```